pkg/config/model/viper.go

```diff
@@ -102,6 +106,11 @@ type safeConfig struct {
 	// keys that have been used but are unknown
 	// used to warn (a single time) on use
 	unknownKeys map[string]struct{}
+
+	// extraConfigFilePaths represents additional configuration file paths that will be merged into the main configuration when ReadInConfig() is called.
+	extraConfigFilePaths []string
+	// extraConfigFilePaths are additional config file paths merged into the configuration during the ReadInConfig() call.
+	loadedExtraConfigFilePaths []string
```

Comment:
Do we need to have 2 slices ? Why not return `extraConfigFilePaths` from `ExtraConfigFilesUsed` ?

Reply:
Imagine we use `extraConfigFilePaths` for two purposes:
1. Store files that will be merge in the Viper instances when `ReadInConfig()` is called
2. Store files that were merged, and return them when `ExtraConfigFilesUsed()` is called

But this solution leads to a tricky case: if one of the extra configuration file is unable to be read or merged, what would we need to do ?
1. remove it from the `extraConfigFilePaths` slice to avoid returning it as used configuration file when `ExtraConfigFilesUsed()` is called ?
2. keep it in the `extraConfigFilePaths` slice in case `ReadInConfig()` is called later to try load it another time ?

But those two solutions leads to inconsistent results:
 - solution 1 modify the internal value of the config instance even in error case
 - solution 2 could return that a file has been loaded while it's wrong

Reply:
As we said it's no longer necessary. I removed in 300a216a812e39b3741cce747de41e873adf39b1.